app/code/Magento/Catalog/Api/ProductRepositoryInterface.php

```diff
@@ -59,13 +59,24 @@ public function delete(\Magento\Catalog\Api\Data\ProductInterface $product);
 
     /**
      * @param string $sku
-     * @return bool Will returned True if deleted
+     * @return bool Will return True if deleted
      * @throws \Magento\Framework\Exception\NoSuchEntityException
      * @throws \Magento\Framework\Exception\StateException
+     * @deprecated Use deleteBySky() to delete a product by SKU
      */
     public function deleteById($sku);
 
     /**
+     * Delete product by SKU
+     *
+     * @param string $sku
+     * @return bool Will return True if deleted
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     * @throws \Magento\Framework\Exception\StateException
+     */
+    public function deleteBySku($sku);
```

Comment:
Unfortunately this is still backward incompatible change as if exists other implementation of this interface it will be broken after new method addition.